### DISCUSSION TITLE
When there is nthRoot in the argument of nthRoot, it becomes a SymbolNode.

### DISCUSSION THREAD

**Discussion by fed-gren** [josdejong/mathjs#2654] [Q&A] [answered]
Hi. I'm running some tests to use the mathjs.

I want to render custom HTML using the handler option with toString or toHTML.

I wrote the code below to express the exponent in nthRoot or power.

```javascript
function customHtml(node, options) {
  if (node.type === "FunctionNode" && node.fn.name === "nthRoot") {
    return `<sup>${node.args[1]}</sup>√${node.args[0]}`;
  } else if(node.type === "OperatorNode" && node.op === "^") {
    return `${node.args[0]}<sup>${node.args[1]}</sup>`;
  }
}
...
const node = math.parse(input);  //input is expression string
const html = node.toString({
  handler: customHtml,
});
```

In the code above, when input is `nthRoot(1, 2)`, I expect html to be `<sup>2</sup>√1`. This works fine.

But when input is `nthRoot(1,(nthRoot(1,2)))`, the result is: `<sup>(nthRoot(1, 2))</sup>√1</div>`

The internal nthRoot is recognized as a **SymbolNode**. Why isn't it recognized as a **FunctionNode**?

**Comment by josdejong** [comment, marked as answer]
You're having for example `node.args[1]` when creating your string, which will simply call the default `.toString()` on your arguments. That should be `node.args[1].toString(options)`, passing your `customHtml` handler via the `options` to nested arguments.
